.github/workflows/developer_onboarding_notification.yml

```diff
@@ -0,0 +1,64 @@
+
+name: PR Merge Comment
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions: write-all
+
+jobs:
+  comment-on-merge:
+    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == 'develop'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: ./.github/actions/merge
```

Comment:
@HardikGoyal2003 Re https://github.com/oppia/oppia/pull/20382#discussion_r1679570147, fair enough, but then should we give this step a name to explain what it actually does, just like the other steps?

Reply:
Hey @U8NWXD Please can you explain, I looked at the code but couldn't get it.

Reply:
I didn't leave this comment, but if you're asking how to give the step a name, take a look at the immediately preceding step:

```yml
      - name: Checkout repository
        uses: actions/checkout@v4
```

`Checkout repository` is the name. If you're confused by the YAML syntax, here's the spec: https://yaml.org/spec/1.2.2/

Reply:
@U8NWXD No, I think I was unclear, what I wanted to ask is what `` - uses: ./.github/actions/merge`` this does.

Reply:
Here's the code for reference:

https://github.com/oppia/oppia/blob/0e6d986205c7e3f4e58ac1f3211358f96e5ae623/.github/actions/merge/action.yml#L1-L26

The branch that the PR was opened against (usually `develop`) is called the `base`. The branch that has the changes in the PR (in this case, `onboarding`) is called the `source`. This action checks out the base branch and merges in the source. This mimics what would happen when the PR is merged. It then leaves the merge commit checked-out so that all the tests run against this simulated post-merge code.

We added this step so that if we fixed flaky tests in develop, PRs would automatically incorporate those fixes as soon as the tests were rerun. Otherwise, PR authors would have to manually merge with develop and rerun all the tests.

Reply:
Ohh Okay!! Got that, Thanks!

Reply:
@HardikGoyal2003 So, are you still going to give this step a name or not? (Do you need it for this workflow?)

Reply:
@seanlip I attempted to name this step, but linters prevented me. I've also reviewed other workflow files where a similar approach was used.

Reply:
Can you share what the linter error was?

Also, I don't think you answered my second question. Why is this step needed for the purposes of your workflow?

Reply:
Hey @seanlip This is the same error as in this [comment](https://github.com/oppia/oppia/pull/20382#discussion_r1679570147) .

I think this step has been mandated for all workflows, else there is no other reason to add this to my workflow.

Reply:
@HardikGoyal2003 -- ah, thanks. Do you know where the linter error originates and which PR introduced that check? I ask because this might be the first workflow we've written that doesn't actually need it, so we might want to consider adding it as an exception to that lint check rather than putting code in here that doesn't make sense.

Reply:
@seanlip This [PR](https://github.com/oppia/oppia/pull/13700) introduced it. and this is the [function](https://github.com/oppia/oppia/blob/9533e555049965e9be16bb507ae7c24161eb13ef/scripts/linters/other_files_linter.py#L317). Thanks!

Reply:
OK, thanks @HardikGoyal2003. I think, let's leave a comment in this step to explain it's not actually needed for this workflow, but is mandated by the lint check _check_that_workflow_steps_use_merge_action() (and point to PR 13700). I'll talk to @U8NWXD about whether we should do anything long-term here. Thanks!

Reply:
@HardikGoyal2003 I talked to @U8NWXD about this, and there is actually a constant [here](
https://github.com/oppia/oppia/blob/9533e555049965e9be16bb507ae7c24161eb13ef/scripts/linters/other_files_linter.py#L71) that you can put this workflow in to exempt it from the lint check in question.

So I want to amend my suggestion in the previous comment -- please instead add your workflow to that list, and drop the step altogether if this workflow does not need it. Thanks!

Reply:
Done!